smooth/framework/functions/functions.py

```diff
@@ -42,3 +42,119 @@ def get_sim_time_span(n_interval, step_size):
     return n_interval * step_size
 
 
+def extract_flow_per_bus(smooth_result):
+    """
+    Extract dict containing the busses that will be plotted.
+    """
+    # Creates empty dict which will later contain the busses that will be plotted.
+    busses_to_plot = dict()
+    nb_trailing_none = 0
+
+    for component_result in smooth_result:
+        if hasattr(component_result, 'flows'):
+            # Track the flows of this component.
+            this_comp_flows = dict()
+            component_flows = component_result.flows
+            for flow in component_flows:
+                # Get rid of "'flow: ".
+                this_flow_name = flow[6:]
+                this_flow_name_split = this_flow_name.split('-->')
+                # Identify the number of trailing None values in case the optimization stopped before termination
+                nb_intervals = len(component_flows[flow])
+                nb_trailing_none = nb_intervals
+                for flow_val in component_flows[flow]:
+                    if flow_val is not None:
+                        nb_trailing_none -= 1
+                    else:
+                        break
+                # check if it's a chp component which consists of two oemof models
+                # if so get rid of the ending '_electric' or '_thermal'
+                if this_flow_name_split[0][-9:] == '_electric':
```

Comment:
Have a look at the endswith() function of strings.

Reply:
I did not change the function, as in the discription said, its only about exporting the functionality not about optimizing it

Reply:
https://github.com/rl-institut/smooth/pull/77/commits/6f8f2544e909f93210dd6a3a4505b1174a4b3488